pkg/nsx/services/vpc/builder.go

```diff
@@ -70,9 +71,8 @@ func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc v1alpha1.VPC
 		}
 	} else {
 		// for creating vpc case, fill in vpc properties based on networkconfig
-		vpcName := util.GenerateIDByObjectByLimit(obj, common.MaxSubnetNameLength)
-		vpc.DisplayName = &vpcName
-		vpc.Id = common.String(util.GenerateIDByObject(obj))
+		vpc.DisplayName = common.String(s.buildVpcName(obj))
+		vpc.Id = common.String(common.BuildUniqueIDWithRandomUUID(obj, util.GenerateIDByObject, s.nsxVpcIdExists))
```

Comment:
seemly, we use networkinfo CR uid instead of namespace uid?

Reply:
updated.